### PR TITLE
WEB: Added theme dependent favicon 

### LINKED ISSUE #42314: WEB: Use appropriate favicon when user has browser in dark mode
We've got two versions of our logo, one for light backgrounds, and another one for dark backgrounds: https://pandas.pydata.org/about/citing.html#brand-and-logo.

Modern browsers usually support a dark mode, and when users have it activated, our favicon (our logo for light backgrounds) doesn't look really nice:

Screenshot at 2021-06-29 13-18-05

Looks like there are ways to change the favicon colors (or the favicon itself I guess) based on the light theme of the browser user. For example, see this article: https://catalin.red/svg-favicon-light-dark-theme/

It'd be nice to show the dark background logo as a favicon to web visitors using a dark theme for their browser.

### PR DESCRIPTION
**Pull request by DeviousLab**
- [x] closes #42314
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Added a light mode version of the favicon based upon the .svgs in the static folder as well as added a small js scipt to the template page to detect the theme change.

mgCmqcreiy

### CODE REVIEW

File: web/pandas/_templates/layout.html
Comment (attack68) [inline web/pandas/_templates/layout.html:130]:
```suggestion
        }

        setupIcons();
```

File: web/pandas/_templates/layout.html
Comment (attack68) [inline web/pandas/_templates/layout.html:105]:
```suggestion
            // change the favicon based on browser detected color scheme on refresh
            const lightSchemeIcon = document.querySelector('link#light-scheme-icon');
```

**Review by attack68 — APPROVED** [review]
lgtm.

**Comment by jreback** [comment]
cc @datapythonista if any comment

**Review by datapythonista — CHANGES_REQUESTED** [review]
Thanks for the contribution @DeviousLab. Looks fine, but I think we can get a way simpler implementation. Your code is only executed at load time, so why not leave the html as is (just adding an id), and the javascript simply have an if to check if the theme is dark, and if it change the href value?

**Comment by DeviousLab** [comment]
@datapythonista I probably have done it this way in the first place, but I've made the changes you've requested, now updates the favicon without requiring refresh through use of a event listener. And now just updates the href of the current favicon. Definitely a lot simpler. 
363GRl3Xq6

**Review by datapythonista — APPROVED** [review]
I don't think a theme update is something we need to worry about, but in any case, this is much clearer, looks good to me.

Thanks for working on it @DeviousLab

**Comment by DeviousLab** [comment]
Are there any more changes that need to be made? Or can it be safely merged without conflicts?

**Comment by datapythonista** [comment]
Sorry for the delay @DeviousLab. We usually require two core devs to review a PR before merging, let's wait a bit more and see if someone else can have a look. Thanks for your patience.

**Comment by DeviousLab** [comment]
> Sorry for the delay @DeviousLab. We usually require two core devs to review a PR before merging, let's wait a bit more and see if someone else can have a look. Thanks for your patience.

No worries! just thought I might have been missing something

**Comment by DeviousLab** [comment]
@jreback Does this look good to you?

**Comment by jreback** [comment]
thanks @DeviousLab

**Pull request merged**
